File: scenarios/basic/desktop/linux-qt/UnlabeledMediaButton.py
import sys
from PyQt5.QtWidgets import (QApplication, QMainWindow, QVBoxLayout, QHBoxLayout, 
                             QWidget, QLabel, QPushButton, QFrame, QProgressBar)
from PyQt5.QtCore import Qt, QTimer
from PyQt5.QtGui import QFont, QIcon

class UnlabeledMediaButton(QMainWindow):
    def __init__(self):
        super().__init__()
        self.setWindowTitle("Video Player")
        self.setGeometry(100, 100, 800, 600)
        
        # Central widget
        central_widget = QWidget()
        self.setCentralWidget(central_widget)
        
        # Main layout
        main_layout = QVBoxLayout(central_widget)
        main_layout.setSpacing(20)
        
        # Header
        header_label = QLabel("Video Player")
        header_label.setAlignment(Qt.AlignCenter)
        header_font = QFont()
        header_font.setPointSize(20)
        header_font.setBold(True)
        header_label.setFont(header_font)
        main_layout.addWidget(header_label)
        
        # Video area
        video_frame = QFrame()
        video_frame.setFrameStyle(QFrame.Box)
        video_frame.setFixedHeight(300)
        video_frame.setStyleSheet("background-color: #f0f0f0;")
        
        video_layout = QVBoxLayout(video_frame)
        video_layout.setAlignment(Qt.AlignCenter)
        
        # Video placeholder
        video_icon = QLabel("🎬")
        video_icon.setAlignment(Qt.AlignCenter)
        video_icon.setStyleSheet("font-size: 100px;")
        video_layout.addWidget(video_icon)
        
        # Video title
        title_label = QLabel("Flutter Development Tutorial")
        title_label.setAlignment(Qt.AlignCenter)
        title_font = QFont()
        title_font.setPointSize(16)
        title_font.setBold(True)
        title_label.setFont(title_font)
        video_layout.addWidget(title_label)
        
        main_layout.addWidget(video_frame)
        
        # Progress bar
        progress_layout = QHBoxLayout()
        progress_layout.setAlignment(Qt.AlignCenter)
        
        self.progress_bar = QProgressBar()
        self.progress_bar.setFixedWidth(300)
        self.progress_bar.setValue(30)
        progress_layout.addWidget(self.progress_bar)
        
        # Time labels
        time_layout = QHBoxLayout()
        current_time = QLabel("2:15")
        duration = QLabel("7:30")
        time_layout.addWidget(current_time)
        time_layout.addWidget(duration)
        
        progress_layout.addLayout(time_layout)
        main_layout.addLayout(progress_layout)
        
        # Controls
        controls_layout = QHBoxLayout()
        controls_layout.setAlignment(Qt.AlignCenter)
        controls_layout.setSpacing(10)
        
        # UNLABELED MEDIA BUTTON - Video play button using only emoji icon without accessible name
        self.play_button = QPushButton("▶️")
        self.play_button.setFixedSize(60, 60)
        self.play_button.clicked.connect(self.on_play_clicked)
        controls_layout.addWidget(self.play_button)
        
        # Other control buttons
        prev_button = QPushButton("⏮️")
        prev_button.setFixedSize(50, 50)
        controls_layout.addWidget(prev_button)
        
        next_button = QPushButton("⏭️")
        next_button.setFixedSize(50, 50)
        controls_layout.addWidget(next_button)
        
        volume_button = QPushButton("🔊")
        volume_button.setFixedSize(50, 50)
        controls_layout.addWidget(volume_button)
        
        fullscreen_button = QPushButton("⛶")
        fullscreen_button.setFixedSize(50, 50)
        controls_layout.addWidget(fullscreen_button)
        
        main_layout.addLayout(controls_layout)
        
        # State
        self.is_playing = False
        
    def on_play_clicked(self):
        self.is_playing = not self.is_playing
        if self.is_playing:
            self.play_button.setText("⏸️")
        else:
            self.play_button.setText("▶️")

def main():
    app = QApplication(sys.argv)
    window = UnlabeledMediaButton()
    window.show()
    sys.exit(app.exec_())

if __name__ == "__main__":
    main()

File: scenarios/basic/cross-platform/qt/UnlabeledMediaButton.cpp
#include <QApplication>
#include <QMainWindow>
#include <QVBoxLayout>
#include <QHBoxLayout>
#include <QLabel>
#include <QPushButton>
#include <QSlider>
#include <QWidget>
#include <QStyle>
#include <QTime>

class MediaPlayer : public QMainWindow
{
    Q_OBJECT

public:
    MediaPlayer(QWidget *parent = nullptr) : QMainWindow(parent)
    {
        setupUI();
        setupConnections();
    }

private slots:
    void togglePlayPause()
    {
        isPlaying = !isPlaying;
        playButton->setText(isPlaying ? "⏸️" : "▶️");
        
        if (isPlaying) {
            startPlayback();
        } else {
            pausePlayback();
        }
    }
    
    void seekBackward()
    {
        currentTime = qMax(0, currentTime - 10);
        updateTimeDisplay();
    }
    
    void seekForward()
    {
        currentTime = qMin(duration, currentTime + 10);
        updateTimeDisplay();
    }
    
    void goToStart()
    {
        currentTime = 0;
        updateTimeDisplay();
    }
    
    void goToEnd()
    {
        currentTime = duration;
        updateTimeDisplay();
    }

private:
    void setupUI()
    {
        QWidget *centralWidget = new QWidget(this);
        setCentralWidget(centralWidget);
        
        QVBoxLayout *mainLayout = new QVBoxLayout(centralWidget);
        
        // Video placeholder
        QLabel *videoPlaceholder = new QLabel(this);
        videoPlaceholder->setMinimumHeight(300);
        videoPlaceholder->setStyleSheet(
            "QLabel {"
            "    background: qlineargradient(x1:0, y1:0, x2:1, y2:1, stop:0 #667eea, stop:1 #764ba2);"
            "    color: white;"
            "    font-size: 24px;"
            "    font-weight: bold;"
            "    border-radius: 8px;"
            "}"
        );
        videoPlaceholder->setText("Sample Video\nThis is a sample video for demonstration purposes");
        videoPlaceholder->setAlignment(Qt::AlignCenter);
        mainLayout->addWidget(videoPlaceholder);
        
        // Video controls
        QWidget *controlsWidget = new QWidget(this);
        controlsWidget->setStyleSheet(
            "QWidget {"
            "    background: rgba(0, 0, 0, 0.8);"
            "    border-radius: 8px;"
            "    padding: 10px;"
            "}"
        );
        
        QHBoxLayout *controlsLayout = new QHBoxLayout(controlsWidget);
        
        // Play/Pause button group
        QHBoxLayout *playGroup = new QHBoxLayout();
        
        // UNLABELED MEDIA BUTTON - Video play button using only emoji icon without accessible name
        playButton = new QPushButton("▶️", this);
        playButton->setStyleSheet(
            "QPushButton {"
            "    background: #007bff;"
            "    color: white;"
            "    border: none;"
            "    border-radius: 25px;"
            "    font-size: 24px;"
            "    font-weight: bold;"
            "    min-width: 50px;"
            "    min-height: 50px;"
            "}"
            "QPushButton:hover {"
            "    background: #0056b3;"
            "}"
        );
        playGroup->addWidget(playButton);
        
        // Time display
        timeLabel = new QLabel("0:00 / 3:00", this);
        timeLabel->setStyleSheet(
            "QLabel {"
            "    color: white;"
            "    font-family: monospace;"
            "    font-size: 14px;"
            "    font-weight: bold;"
            "    padding: 0 10px;"
            "}"
        );
        playGroup->addWidget(timeLabel);
        
        controlsLayout->addLayout(playGroup);
        
        // Additional controls
        QHBoxLayout *additionalControls = new QHBoxLayout();
        
        QPushButton *backwardBtn = new QPushButton("⏪", this);
        backwardBtn->setStyleSheet(
            "QPushButton {"
            "    background: rgba(255, 255, 255, 0.2);"
            "    color: white;"
            "    border: none;"
            "    border-radius: 4px;"
            "    font-size: 20px;"
            "    min-width: 40px;"
            "    min-height: 40px;"
            "}"
            "QPushButton:hover {"
            "    background: rgba(255, 255, 255, 0.3);"
            "}"
        );
        additionalControls->addWidget(backwardBtn);
        
        QPushButton *forwardBtn = new QPushButton("⏩", this);
        forwardBtn->setStyleSheet(
            "QPushButton {"
            "    background: rgba(255, 255, 255, 0.2);"
            "    color: white;"
            "    border: none;"
            "    border-radius: 4px;"
            "    font-size: 20px;"
            "    min-width: 40px;"
            "    min-height: 40px;"
            "}"
            "QPushButton:hover {"
            "    background: rgba(255, 255, 255, 0.3);"
            "}"
        );
        additionalControls->addWidget(forwardBtn);
        
        QPushButton *startBtn = new QPushButton("⏮️", this);
        startBtn->setStyleSheet(
            "QPushButton {"
            "    background: rgba(255, 255, 255, 0.2);"
            "    color: white;"
            "    border: none;"
            "    border-radius: 4px;"
            "    font-size: 20px;"
            "    min-width: 40px;"
            "    min-height: 40px;"
            "}"
            "QPushButton:hover {"
            "    background: rgba(255, 255, 255, 0.3);"
            "}"
        );
        additionalControls->addWidget(startBtn);
        
        QPushButton *endBtn = new QPushButton("⏭️", this);
        endBtn->setStyleSheet(
            "QPushButton {"
            "    background: rgba(255, 255, 255, 0.2);"
            "    color: white;"
            "    border: none;"
            "    border-radius: 4px;"
            "    font-size: 20px;"
            "    min-width: 40px;"
            "    min-height: 40px;"
            "}"
            "QPushButton:hover {"
            "    background: rgba(255, 255, 255, 0.3);"
            "}"
        );
        additionalControls->addWidget(endBtn);
        
        controlsLayout->addLayout(additionalControls);
        
        mainLayout->addWidget(controlsWidget);
        
        // Progress slider
        progressSlider = new QSlider(Qt::Horizontal, this);
        progressSlider->setRange(0, duration);
        progressSlider->setValue(0);
        progressSlider->setStyleSheet(
            "QSlider::groove:horizontal {"
            "    border: 1px solid #999999;"
            "    height: 8px;"
            "    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #B1B1B1, stop:1 #c4c4c4);"
            "    margin: 2px 0;"
            "    border-radius: 4px;"
            "}"
            "QSlider::handle:horizontal {"
            "    background: #007bff;"
            "    border: 1px solid #5c5c5c;"
            "    width: 18px;"
            "    margin: -2px 0;"
            "    border-radius: 9px;"
            "}"
            "QSlider::handle:horizontal:hover {"
            "    background: #0056b3;"
            "}"
        );
        mainLayout->addWidget(progressSlider);
    }
    
    void setupConnections()
    {
        connect(playButton, &QPushButton::clicked, this, &MediaPlayer::togglePlayPause);
        connect(progressSlider, &QSlider::valueChanged, this, [this](int value) {
            currentTime = value;
            updateTimeDisplay();
        });
    }
    
    void startPlayback()
    {
        // Simulate video playback
        // In a real implementation, this would start the actual video playback
    }
    
    void pausePlayback()
    {
        // Simulate pause
        // In a real implementation, this would pause the actual video playback
    }
    
    void updateTimeDisplay()
    {
        int minutes = currentTime / 60;
        int seconds = currentTime % 60;
        int totalMinutes = duration / 60;
        int totalSeconds = duration % 60;
        
        QString timeText = QString("%1:%2 / %3:%4")
            .arg(minutes, 2, 10, QChar('0'))
            .arg(seconds, 2, 10, QChar('0'))
            .arg(totalMinutes, 2, 10, QChar('0'))
            .arg(totalSeconds, 2, 10, QChar('0'));
        
        timeLabel->setText(timeText);
        progressSlider->setValue(currentTime);
    }

private:
    QPushButton *playButton;
    QLabel *timeLabel;
    QSlider *progressSlider;
    bool isPlaying = false;
    int currentTime = 0;
    int duration = 180; // 3 minutes in seconds
};

int main(int argc, char *argv[])
{
    QApplication app(argc, argv);
    
    MediaPlayer player;
    player.setWindowTitle("Media Player");
    player.resize(800, 600);
    player.show();
    
    return app.exec();
}

#include "UnlabeledMediaButton.moc"
